cmake/Findopentelemetry.cmake

```diff
@@ -118,5 +119,6 @@ if(opentelemetry_FOUND)
         list(APPEND opentelemetry_LIBRARIES opentelemetry::${OTLIB})
       endif()
     endforeach()
+    target_compile_options(opentelemetry::opentelemetry_trace INTERFACE -Wno-cpp)
```

Comment:
`opentelemetry::opentelemetry` or `opentelemetry::opentelemetry_trace` ?

Reply:
The trace is the specific component that had the problem, so I applied it there. I wasn't 100% sure that `opentelemetry::opentelemetry` would work for the non-Apple case.

Reply:
Yeah, I tested `opentelemetry::opentelemetry_trace` and that's not actually defined for the non-Apple condition in here. But `opentelemetry::opentelemetry_trace` does make the build happy.